packages/cloud-functions/src/cico/Moonpay.ts

```diff
@@ -0,0 +1,144 @@
+import { DigitalAsset, FETCH_TIMEOUT_DURATION, MOONPAY_DATA } from '../config'
+import { PaymentMethod, ProviderQuote, UserLocationData } from './fetchProviders'
+import { bankingSystemToCountry } from './providerAvailability'
+import { fetchLocalCurrencyAndExchangeRate, fetchWithTimeout } from './utils'
+
+interface MoonpayQuote {
+  baseCurrency: {
+    id: string
+    createdAt: string
+    updatedAt: string
+    type: string
+    name: string
+    code: string
+    precision: number
+  }
+  baseCurrencyAmount: number
+  currency: {
+    id: string
+    createdAt: string
+    updatedAt: string
+    type: string
+    name: string
+    code: string
+    precision: number
+    addressRegex: string
+    testnetAddressRegex: string
+    minAmount: number
+    maxAmount: number
+    supportsAddressTag: boolean
+    addressTagRegex: string | null
+    supportsTestMode: boolean
+    isSuspended: boolean
+    isSupportedInUS: boolean
+    isSellSupported: boolean
+    notAllowedUSStates: string[]
+  }
+  extraFeeAmount: number
+  feeAmount: number
+  networkFeeAmount: number
+  paymentMethod: string
+  quoteCurrencyAmount: number
+  totalAmount: number
+}
+
+const Moonpay = {
+  fetchQuote: async (
+    digitalAsset: DigitalAsset,
+    fiatCurrency: string,
+    fiatAmount: number | undefined,
+    userLocation: UserLocationData
+  ) => {
+    try {
+      if (!fiatAmount) {
+        throw Error('Purchase amount not provided')
+      }
+
+      let { localCurrency, exchangeRate } = await fetchLocalCurrencyAndExchangeRate(
+        userLocation.country,
+        fiatCurrency
+      )
+
+      // If the local currency is not supported by Moonpay, then get estimate in USD
+      if (!MOONPAY_DATA.supported_currencies.includes(localCurrency)) {
+        ;({ localCurrency, exchangeRate } = await fetchLocalCurrencyAndExchangeRate(
+          userLocation.country,
+          fiatCurrency,
+          'USD'
+        ))
+      }
+
+      const localFiatAmount = fiatAmount * exchangeRate
+
+      const paymentMethods = ['credit_debit_card']
+      if (userLocation.country) {
+        if (bankingSystemToCountry.gbp[userLocation.country]) {
+          paymentMethods.push('gbp_bank_transfer')
+        } else if (bankingSystemToCountry.sepa[userLocation.country]) {
+          paymentMethods.push('sepa_bank_transfer')
+        }
+      }
+
+      const baseUrl = `
+        ${MOONPAY_DATA.api_url}
+        /v3
+        /currencies
+        /${digitalAsset.toLowerCase()}
+        /buy_quote
+        /?apiKey=${MOONPAY_DATA.public_key}
+        &baseCurrencyCode=${localCurrency.toLowerCase()}
+        &baseCurrencyAmount=${localFiatAmount.toFixed(2)}
+      `.replace(/\s+/g, '')
```

Comment:
This is primarily in for the new line spaces I've put in the template literal. We don't actually want new lines for each input in the final string but I've found it nice for readability

Reply:
ohhhh, makes sense.

Reply:
Now that I see the explanation that's really clever, how would you feel about moving the regex to a constant so that we can tell what it's doing by its name?